samples/nrf9160/nrf_cloud_rest_fota/src/main.c

```diff
@@ -0,0 +1,826 @@
+/* Buffer used for REST calls */
+static char rx_buf[REST_RX_BUF_SZ];
+
+#define JWT_DURATION_S	(60*5)
+#define JWT_BUF_SZ	900
+/* Buffer used for JSON Web Tokens (JWTs) */
+static char jwt[JWT_BUF_SZ];
+
+/* nRF Cloud REST context */
+struct nrf_cloud_rest_context rest_ctx = {
+	.connect_socket = -1,
+	.keep_alive = true,
+	.rx_buf = rx_buf,
+	.rx_buf_len = sizeof(rx_buf),
+	.fragment_size = 0
+};
+
+#if defined(CONFIG_REST_FOTA_DO_JITP)
+/* Flag to indicate if the user requested JITP to be performed */
+static bool jitp_requested;
+#endif
+
+/* Flag to indicate if FOTA should be enabled in the device shadow */
+static bool enable_fota_requested;
+
+static int rest_fota_settings_set(const char *key, size_t len_rd,
+			    settings_read_cb read_cb, void *cb_arg);
+
+SETTINGS_STATIC_HANDLER_DEFINE(rest_fota, FOTA_SETTINGS_NAME, NULL,
+			       rest_fota_settings_set, NULL, NULL);
+
+static int rest_fota_settings_set(const char *key, size_t len_rd, settings_read_cb read_cb,
+				  void *cb_arg)
+{
+	if (!key) {
+		LOG_DBG("Key is NULL");
+		return -EINVAL;
+	}
+
+	LOG_DBG("Settings key: %s, size: %d", log_strdup(key), len_rd);
+
+	if (!strncmp(key, FOTA_SETTINGS_KEY_PENDING_JOB, strlen(FOTA_SETTINGS_KEY_PENDING_JOB)) &&
+	    (len_rd == sizeof(pending_job))) {
+
+		if (read_cb(cb_arg, (void *)&pending_job, len_rd) == len_rd) {
+
+			LOG_DBG("Saved job: %s, type: %d, validate: %d",
+				log_strdup(pending_job.id),
+				pending_job.type,
+				pending_job.validate);
+			return 0;
+		}
+	}
+	return -ENOTSUP;
+}
+
+static void http_fota_dl_handler(const struct fota_download_evt *evt)
+{
+	LOG_DBG("evt: %d", evt->id);
+
+	switch (evt->id) {
+	case FOTA_DOWNLOAD_EVT_FINISHED:
+		LOG_INF("FOTA download finished");
+		fota_status = NRF_CLOUD_FOTA_SUCCEEDED;
+		k_sem_give(&fota_download_sem);
+		break;
+	case FOTA_DOWNLOAD_EVT_ERASE_PENDING:
+		LOG_INF("FOTA download erase pending");
+		fota_status = NRF_CLOUD_FOTA_SUCCEEDED;
+		k_sem_give(&fota_download_sem);
+		break;
+	case FOTA_DOWNLOAD_EVT_ERASE_DONE:
+		LOG_DBG("FOTA download erase done");
+		break;
+	case FOTA_DOWNLOAD_EVT_ERROR:
+		LOG_INF("FOTA download error: %d", evt->cause);
+
+		fota_status = NRF_CLOUD_FOTA_FAILED;
+		fota_status_details = FOTA_STATUS_DETAILS_DL_ERR;
+
+		if (evt->cause == FOTA_DOWNLOAD_ERROR_CAUSE_INVALID_UPDATE) {
+			fota_status = NRF_CLOUD_FOTA_REJECTED;
+			if (job.type == NRF_CLOUD_FOTA_MODEM) {
+				fota_status_details = FOTA_STATUS_DETAILS_MDM_REJ;
+			} else {
+				fota_status_details = FOTA_STATUS_DETAILS_MCU_REJ;
+			}
+		} else if (evt->cause == FOTA_DOWNLOAD_ERROR_CAUSE_TYPE_MISMATCH) {
+			fota_status_details = FOTA_STATUS_DETAILS_MISMATCH;
+		}
+		k_sem_give(&fota_download_sem);
+		break;
+	case FOTA_DOWNLOAD_EVT_PROGRESS:
+		LOG_INF("FOTA download percent: %d", evt->progress);
+		break;
+	default:
+		break;
+	}
+}
+
+int set_led(const int state)
+{
+#if defined(CONFIG_REST_FOTA_ENABLE_LED)
+#if defined(CONFIG_BOARD_CIRCUITDOJO_FEATHER_NRF9160NS) || \
```

Comment:
Not sure about this, do we now support some 3rd party boards in our samples?

Reply:
probably not... i was just using it for testing.  i'll remove.